### PR TITLE
Add http_auth_request fuzzer

### PR DESCRIPTION
**Pull request by PaDarochek**

### CODE REVIEW

File: projects/nginx/Dockerfile [hunk 1]
```diff
@@ -16,12 +16,53 @@
...
-RUN hg clone http://hg.nginx.org/nginx/ -u d1b8568f3042
+# ansifilter
+ARG GIT_ANSIFILTER_TAG="2.18"
+ARG ANSIFILTER_SRC_DIR="/ansifilter"
+RUN : \
+    && git clone --depth=1 https://gitlab.com/saalen/ansifilter.git \
+        -b ${GIT_ANSIFILTER_TAG} ${ANSIFILTER_SRC_DIR} \
+    && cd ${ANSIFILTER_SRC_DIR} \
+    && env CC=clang CXX=clang++ make \
+    && make install \
```
Comment (headshog) [inline projects/nginx/Dockerfile:38]: Maybe it is better to move installation step to build.sh?

File: projects/nginx/Dockerfile [hunk 2]
```diff
@@ -16,12 +16,53 @@
...
+    && git config --global advice.detachedHead false \
+    && :
+
+        # Clone target from GitHub. Checkout specified commit. It could be updated later.
+RUN hg clone http://hg.nginx.org/nginx/ -u d1b8568f3042 /nginx
+
+RUN : \
+    && git clone https://github.com/zardus/preeny.git \
+    && cd preeny \
+    && make
```
Comment (headshog) [inline projects/nginx/Dockerfile:65]: Move to buils.sh?

File: projects/nginx/Dockerfile [hunk 3]
```diff
@@ -16,12 +16,53 @@
...
+# Settings
+RUN : \
+    && echo -e "set -g mouse on\nset -g mode-mouse on\n" > /etc/tmux.conf \
+    && sh -c 'echo set encoding=utf-8 > /root/.vimrc' \
+    && echo '. /usr/share/bash-completion/bash_completion' >> ~/.bashrc \
+    && echo "export PS1='"'[fuzzuser \h] \w \$ '"'" >> ~/.bashrc \
+    && git config --global advice.detachedHead false \
+    && :
+
+        # Clone target from GitHub. Checkout specified commit. It could be updated later.
```
Comment (headshog) [inline projects/nginx/Dockerfile:59]: Remove indent

File: projects/nginx/auth_fuzz/build.sh [hunk 4]
```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
```
Comment (headshog) [inline projects/nginx/auth_fuzz/build.sh:1]: Add copyright

File: projects/nginx/auth_fuzz/http_auth_request-afl++.toml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# Copyright 2022 ISP RAS
```
Comment (headshog) [inline projects/nginx/auth_fuzz/http_auth_request-afl++.toml:1]: Update copyright

File: projects/nginx/auth_fuzz/inputs/in1 [hunk 6]
```diff
@@ -0,0 +1 @@
+HEAD / HTTP/1.1
```
Comment (headshog) [inline projects/nginx/auth_fuzz/inputs/in1:1]: Maybe it is better to zip all of these inputs and unzip them in build.sh?

File: projects/nginx/nginx.conf [hunk 7]
```diff
@@ -0,0 +1,120 @@
+daemon off;
```
Comment (headshog) [inline projects/nginx/nginx.conf:1]: Do we need copyright here?

**Review by headshog — CHANGES_REQUESTED** [review]

File: projects/nginx/Dockerfile (hunk 3 above)
Comment (PaDarochek) [reply projects/nginx/Dockerfile:59]: Done

File: projects/nginx/Dockerfile (hunk 1 above)
Comment (PaDarochek) [reply projects/nginx/Dockerfile:38]: Done

File: projects/nginx/Dockerfile (hunk 2 above)
Comment (PaDarochek) [reply projects/nginx/Dockerfile:65]: Done

File: projects/nginx/auth_fuzz/build.sh (hunk 4 above)
Comment (PaDarochek) [reply projects/nginx/auth_fuzz/build.sh:1]: Done

File: projects/nginx/auth_fuzz/http_auth_request-afl++.toml (hunk 5 above)
Comment (PaDarochek) [reply projects/nginx/auth_fuzz/http_auth_request-afl++.toml:1]: Done

File: projects/nginx/auth_fuzz/inputs/in1 (hunk 6 above)
Comment (PaDarochek) [reply projects/nginx/auth_fuzz/inputs/in1:1]: Done

File: projects/nginx/nginx.conf (hunk 7 above)
Comment (PaDarochek) [reply projects/nginx/nginx.conf:1]: No, as is request_fuzz/http_config.conf

File: projects/nginx/Dockerfile [hunk 8]
```diff
@@ -16,34 +16,68 @@
...
+RUN apt-get update \
+    && apt-get dist-upgrade -y \
+    && apt-get install -y \
+    libpcre3-dev zlib1g-dev mercurial ninja-build liblzma-dev \
+    libz-dev binutils libtool build-essential libssl-dev libseccomp-dev \
+    libini-config-dev \
+    && apt-get autoclean
+
+# ansifilter
+ARG GIT_ANSIFILTER_TAG="2.18"
```
Comment (headshog) [inline projects/nginx/Dockerfile:29]: This arg can be removed, it is used only one time below

File: projects/nginx/Dockerfile [hunk 9]
```diff
@@ -16,34 +16,68 @@
...
+ARG ANSIFILTER_SRC_DIR="/ansifilter"
```
Comment (headshog) [inline projects/nginx/Dockerfile:30]: This arg can be removed too

**Review by headshog — CHANGES_REQUESTED** [review]

**Pull request merged**